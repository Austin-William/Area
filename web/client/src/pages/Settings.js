import React from "react";
import axios from "axios";
import { NotificationManager } from 'react-notifications';
import "react-notifications/lib/notifications.css";
import { getAuth, signOut } from "firebase/auth";
import "../styles/Settings.css";
import { url } from "../global/variables";

function Settings(props) {

    const [ip, setIp] = React.useState("");
    const [seconds, setSeconds] = React.useState("");
    const [email, setEmail] = React.useState("");
    const [phone, setPhone] = React.useState("");

    const handleIp = (event) => {
        setIp(event.target.value);
    }

    const handleEmail = (event) => {
        setEmail(event.target.value);
    }

    const handlePhone = (event) => {
        setPhone(event.target.value);
    }

    const handleSecondsTimer = (event) => {
        setSeconds(event.target.value);
    }

    const handleDisconnect = () => {
        const auth = getAuth();
        signOut(auth)
            .then(() => {
                NotificationManager.success("Disconnected", "Success", 3000);
                setEmail("");
                setPhone("");
                window.location.href = "/login";
            })
            .catch(error => {
                console.log(error);
            });
    }

    const handleDeleteAccount = () => {
        const auth = getAuth();
        var confirmDelete = window.confirm("Are you sure you want to delete your account?");

        if (confirmDelete) {
            axios.delete(url + "/api/user/" + props.id);
            auth.currentUser.delete()
            .then(() => {
                window.location.href = "/login";
            })
            .catch(error => {
                console.log(error);
            });
        }
    }

    const handleSubmit = (event) => {
        event.preventDefault();
        if (ip !== "") {
            if (seconds !== 0) {
                try {
                    axios.post(ip + "/api/reactions/timer/" + seconds)
                        .then(function (response) {
                            if (response.status === 200) {
                                NotificationManager.success("Timer successfully set to " + seconds, "Success", 3000);
                            }
                        })
                        .catch(function (error) {
                            NotificationManager.error("Timer not set", "Error", 3000);
                        });
                } catch (error) {
                    console.log(error);
                }
            }
            if (email !== "") {
                try {
                    axios.post(ip + '/api/email/' + email)
                        .then((response) => {
                            if (response.status === 200) {
                                NotificationManager.success("Email successfully set to " + email, "Success", 3000);
                            }
                        })
                        .catch(() => {
                            NotificationManager.error("Email not set", "Error", 3000);
                        });
                } catch (error) {
                    console.log(error);
                }
            }
            if (phone !== "") {
                console.log(phone);
            }
        }
    }

    React.useEffect(() => {
        if (ip === "") {
            setIp(url);
        }
        try {
            const response = axios.get(ip + '/api/email')
                .then((response) => {
                    setEmail(response.data.email);
                })
                .catch(() => {
                    console.log('Request failed');
                });
            if (response.status !== 200) {
                const auth = getAuth();
                setEmail(auth.currentUser.email);
            }
        } catch (error) {
            console.log(error);
        }
    }, [ip, email]);

    return (
        <div className="Settings">
            <div className="Settings-container">
                <div className="Settings-content">
                    <div className="Settings-content-item">
                        <h3>IP</h3>
                        <input type="text" placeholder={ip} onChange={handleIp} />
                    </div>
                    <div className="Settings-content-item">
                        <h3>Email</h3>
                        <input type="text" placeholder={email} onChange={handleEmail} />
                    </div>
                    <div className="Settings-content-item">
                        <h3>Phone</h3>
                        <input type="text" placeholder={phone} onChange={handlePhone} />
                    </div>
                    <div className="Settings-content-item">
                        <h3>Timer</h3>
                        <input type="number" placeholder={seconds} onChange={handleSecondsTimer} />
                    </div>
                </div>
                <button className="Settings-button" onClick={handleSubmit}>
                    Save
                </button>
            </div>
            <div className="Settings-container">
                <button className="Settings-button-disconnect" onClick={handleDisconnect}>
                    Disconnection
                </button>
                <button className="Settings-button-disconnect" onClick={handleDeleteAccount}>
                    Delete your account
                </button>
            </div>
        </div>
    );
}

export default Settings;