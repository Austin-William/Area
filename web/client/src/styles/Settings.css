/* This file change the display of settings toast */

.Settings {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    width: 100%;
    height: 100%;
    color: whitesmoke;
}

.Settings-container {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    width: 100%;
    height: 100%;
    padding: 2em;
    color: whitesmoke;
}

/* Settings's content CSS part */

.Settings-content {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    width: 60%;
    height: 80%;
}

.Settings-content .Settings-content-item {
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: center;
    width: 100%;
    height: 10%;
    margin: 0.5em;
    padding: 0.5em;
    border-radius: 0.5em;
    background-color: whitesmoke;
}

.Settings-content .Settings-content-item h3 {
    font-size: 1.5em;
    margin: 0.5em;
    padding: 0.5em;
    color: grey;
    border-radius: 0.5em;
}

.Settings-content .Settings-content-item input {
    width: 100%;
    height: 50%;
    border-radius: 0.5em;
    font-size: 1.5em;
    margin: 0.5em;
    padding: 0.5em;
    color: grey;
    transition: 0.5s;
}

.Settings-content .Settings-content-item input:hover {
    border-color: crimson;
}

.Settings-container .Settings-button, .Settings-container .Settings-button-disconnect {
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: center;
    width: 20%;
    height: 10%;
    margin: 0.5em;
    padding: 0.5em;
    border-radius: 0.5em;
    color: whitesmoke;
    border-color: blueviolet;
    background-color: transparent;
    transition: 0.3s;
}

.Settings-button-disconnect:hover {
    background-color: red;
}

.Settings-container .Settings-button:hover {
    border-color: whitesmoke;
    background-color: whitesmoke;
    color: blueviolet;
}

/* CSS part for responsive design */

@media screen and (max-width: 768px) {
    .Settings-container {
        width: 100%;
        height: 100%;
        padding: 0.5em;
    }

    .Settings-content .Settings-content-item {
        width: 100%;
    }

    .Settings-content .Settings-content-item h3 {
        font-size: 1.2em;
        color: grey;
    }

    .Settings-content .Settings-content-item input {
        width: 100%;
        font-size: 1em;
        transition: 0.5s;
    }
}