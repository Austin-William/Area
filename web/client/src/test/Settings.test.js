import React from "react";
import renderer from "react-test-renderer";
import Settings from "../pages/Settings";

test("Pages : Settings renders correctly", () => {
  const tree = renderer.create(<Settings />).toJSON();
  expect(tree).toMatchSnapshot();
});

