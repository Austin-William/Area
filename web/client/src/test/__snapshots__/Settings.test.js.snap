// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages : Settings renders correctly 1`] = `
<div
  className="Settings"
>
  <div
    className="Settings-container"
  >
    <div
      className="Settings-content"
    >
      <div
        className="Settings-content-item"
      >
        <h3>
          IP
        </h3>
        <input
          onChange={[Function]}
          placeholder=""
          type="text"
        />
      </div>
      <div
        className="Settings-content-item"
      >
        <h3>
          Email
        </h3>
        <input
          onChange={[Function]}
          placeholder=""
          type="text"
        />
      </div>
      <div
        className="Settings-content-item"
      >
        <h3>
          Phone
        </h3>
        <input
          onChange={[Function]}
          placeholder=""
          type="text"
        />
      </div>
      <div
        className="Settings-content-item"
      >
        <h3>
          Timer
        </h3>
        <input
          onChange={[Function]}
          placeholder=""
          type="number"
        />
      </div>
    </div>
    <button
      className="Settings-button"
      onClick={[Function]}
    >
      Save
    </button>
  </div>
</div>
`;
